src/test/rustdoc-json/impls/auto.rs

```diff
@@ -0,0 +1,10 @@
+#![feature(no_core, auto_traits, lang_items)]
+#![no_core]
+
+#[lang = "sized"]
+trait Sized {}
+
+pub auto trait Bar {}
+
+// @has auto.json "$.index[*][?(@.kind=='impl')].span" null
```

Comment:
This test isn't that specific, as it only checks that their is an impl with an null span. A better one would check that the synthetic impl has a null span, and a real one has the correct span.

(Also, I need to document how the json testing infra works)

```suggestion
pub struct Foo;

/// hack
impl Foo {
    pub fn baz(&self) {}
}

// Testing spans, so all tests below code

// @is auto.json "$.index[*][?(@.kind=='impl' && @.inner.synthetic==true)].span" null

// @is - "$.index[*][?(@.docs=='hack')].span.begin" "[12, 0]"
// @is - "$.index[*][?(@.docs=='hack')].span.end" "[14, 1]"
```